### PR TITLE
Use MoveIt fake controller for the demo

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_demo_nodes/run_moveit_cpp/src/run_moveit_cpp.cpp [hunk 1]
```diff
@@ -63,6 +61,16 @@ class MoveItCppDemo
     moveit_cpp_ = std::make_shared<moveit::planning_interface::MoveItCpp>(node_);
     moveit_cpp_->getPlanningSceneMonitor()->setPlanningScenePublishingFrequency(100);
 
+    // Wait until there are active controllers available
+    while (rclcpp::ok())
+    {
+      if (moveit_cpp_->getTrajectoryExecutionManager()->ensureActiveControllersForGroup("panda_arm"))
```
Comment (henningkayser) [inline moveit_demo_nodes/run_moveit_cpp/src/run_moveit_cpp.cpp:67]: Shouldn't MoveItCpp handle this internally?

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 2]
```diff
@@ -568,14 +568,14 @@ void TrajectoryExecutionManager::updateControllerState(const std::string& contro
 
 void TrajectoryExecutionManager::updateControllerState(ControllerInformation& ci, const rclcpp::Duration& age)
 {
-  if (rclcpp::Clock().now() - ci.last_update_ >= age)
+  if (rclcpp::Clock(RCL_ROS_TIME).now() - ci.last_update_ >= age)
```
Comment (henningkayser) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp:571]: Maybe a better readable way to get the current time is `Node::now()`. By default nodes are initialized with ros time.

File: moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py [hunk 3]
```diff
@@ -79,14 +86,21 @@ def generate_launch_description():
...
-
-    return LaunchDescription([ static_tf, rviz_node, run_moveit_cpp_node, fake_joint_driver_node ])
+    # Joint state publisher
+    joint_state_publisher = Node(package='joint_state_publisher',
+                                 node_executable='joint_state_publisher',
+                                 node_name='joint_state_publisher',
+                                 arguments=[os.path.join(get_package_share_directory('moveit_resources'), 'panda_description/urdf/panda.urdf')],
+                                 output='log',
+                                 parameters=[{'source_list': ['/fake_controller_joint_states'],
+                                              'zeros': {'panda_joint1': 0.0,
```
Comment (henningkayser) [inline moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py:96]: Can't we use the initial pose instead of 'zeros'?

**Comment by henningkayser** [comment]
@JafarAbdi ping

**Comment by henningkayser** [comment]
When I try to run this I get the following exception:
```
[DEBUG] [launch.launch_context]: emitting event synchronously: 'launch.events.IncludeLaunchDescription'
[DEBUG] [launch.launch_context]: emitting event synchronously: 'launch.events.IncludeLaunchDescription'
[INFO] [launch]: All log files can be found below /home/henning/.ros/log/2020-07-22-20-13-45-824339-henning-tp480s-26040
[INFO] [launch]: Default logging verbosity is set to DEBUG
[DEBUG] [launch]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f0a2a4a5470>'
[DEBUG] [launch]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f0a2a4a5470>' ✓ '<launch.event_handlers.on_include_launch_description.OnIncludeLaunchDescription object at 0x7f0a200a55f8>'
[DEBUG] [launch]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f0a1c1960b8>'
[DEBUG] [launch]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f0a1c1960b8>' ✓ '<launch.event_handlers.on_include_launch_description.OnIncludeLaunchDescription object at 0x7f0a200a55f8>'
/home/henning/ros2/moveit2/install/run_moveit_cpp/share/run_moveit_cpp/launch/run_moveit_cpp.launch.py:23: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  return yaml.load(file)
[DEBUG] [launch]: Traceback (most recent call last):
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_description_sources/any_launch_file_utilities.py", line 53, in get_launch_description_from_any_launch_file
    return loader(launch_file_path)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_description_sources/python_launch_file_utilities.py", line 68, in get_launch_description_from_python_launch_file
    return getattr(launch_file_module, 'generate_launch_description')()
  File "/home/henning/ros2/moveit2/install/run_moveit_cpp/share/run_moveit_cpp/launch/run_moveit_cpp.launch.py", line 71, in generate_launch_description
    fake_controller])
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch_ros/actions/node.py", line 150, in __init__
    normalized_params = normalize_parameters(parameters)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch_ros/utilities/normalize_parameters.py", line 181, in normalize_parameters
    normalized_params.append(normalize_parameter_dict(param))
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch_ros/utilities/normalize_parameters.py", line 146, in normalize_parameter_dict
    sub_dict = normalize_parameter_dict(value, _prefix=name)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch_ros/utilities/normalize_parameters.py", line 160, in normalize_parameter_dict
    normalized[tuple(name)] = _normalize_parameter_array_value(value)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch_ros/utilities/normalize_parameters.py", line 69, in _normalize_parameter_array_value
    raise RuntimeError('Failed to handle type {}'.format(repr(subvalue)))
RuntimeError: Failed to handle type {'name': 'fake_panda_arm_controller', 'joints': ['panda_joint1', 'panda_joint2', 'panda_joint3', 'panda_joint4', 'panda_joint5', 'panda_joint6', 'panda_joint7']}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_service.py", line 385, in run_async
    await process_one_event_task
  File "/usr/lib/python3.6/asyncio/coroutines.py", line 126, in send
    return self.gen.send(value)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_service.py", line 295, in _process_one_event
    await self.__process_event(next_event)
  File "/usr/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_service.py", line 315, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/actions/include_launch_description.py", line 125, in execute
    launch_description = self.__launch_description_source.get_launch_description(context)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_description_source.py", line 84, in get_launch_description
    self._get_launch_description(self.__expanded_location)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_description_sources/any_launch_description_source.py", line 53, in _get_launch_description
    return get_launch_description_from_any_launch_file(location)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_description_sources/any_launch_file_utilities.py", line 56, in get_launch_description_from_any_launch_file
    raise InvalidLaunchFileError(extension, likely_errors=exceptions)
launch.invalid_launch_file_error.InvalidLaunchFileError: Caught exception when trying to load file of format [py]: Failed to handle type {'name': 'fake_panda_arm_controller', 'joints': ['panda_joint1', 'panda_joint2', 'panda_joint3', 'panda_joint4', 'panda_joint5', 'panda_joint6', 'panda_joint7']}

[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [py]: Failed to handle type {'name': 'fake_panda_arm_controller', 'joints': ['panda_joint1', 'panda_joint2', 'panda_joint3', 'panda_joint4', 'panda_joint5', 'panda_joint6', 'panda_joint7']}
[DEBUG] [launch.launch_context]: emitting event: 'launch.events.Shutdown'
[DEBUG] [launch]: processing event: '<launch.events.shutdown.Shutdown object at 0x7f0a286f9780>'
[DEBUG] [launch]: processing event: '<launch.events.shutdown.Shutdown object at 0x7f0a286f9780>' ✓ '<launch.event_handlers.on_shutdown.OnShutdown object at 0x7f0a1af1c518>'
[DEBUG] [launch]: processing event: '<launch.events.shutdown.Shutdown object at 0x7f0a286f9780>' ✓ '<launch.event_handlers.on_shutdown.OnShutdown object at 0x7f0a286e7748>'
Executing <Task finished coro=<LaunchService._process_one_event() done, defined at /opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_service.py:293> result=None created at /opt/ros/eloquent/lib/python3.6/site-packages/launch/launch_service.py:363> took 0.951 seconds
```

**Comment by JafarAbdi** [comment]
> When I try to run this I get the following exception:
> ...

Hmm, I just tested it again and it working on my case, I'll try to clean my workspace and try again

File: moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py (hunk 3 above)
Comment (JafarAbdi) [reply moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py:96]: joint_state_publisher keep publishing zero values (the robot is in collision with these joint state) until it gets a message from MoveIt fake controller which takes some time to initialize causing MoveItCpp to get the wrong angles

Comment (henningkayser) [reply moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py:96]: Setting `publish_default_positions=false` should do the trick

**Comment by JafarAbdi** [comment]
This PR should be merge after we use foxy for the main branch -- it depends on https://github.com/JafarAbdi/fake_joint/tree/foxy and https://github.com/ros-planning/moveit_resources/pull/38

**Comment by henningkayser** [comment]
Github didn't get Travis results, but build succeeded without issues (https://travis-ci.com/github/ros-planning/moveit2/jobs/370735189). Merging by hand.

**Pull request merged**